src/wallet/wallet.cpp

```diff
@@ -508,30 +508,39 @@ void CWallet::SetBestChain(const CBlockLocator& loc)
     SetBestChainINTERNAL(walletdb, loc);
 }
 
-std::set<std::pair<libzcash::PaymentAddress, uint256>> CWallet::GetNullifiersForAddresses(const std::set<libzcash::PaymentAddress> & addresses)
+std::set<std::pair<libzcash::PaymentAddress, uint256>> CWallet::GetNullifiersForAddresses(
+        const std::set<libzcash::PaymentAddress> & addresses)
 {
     std::set<std::pair<libzcash::PaymentAddress, uint256>> nullifierSet;
-    // Sapling ivk -> addr map
-    std::map<libzcash::SaplingIncomingViewingKey, libzcash::SaplingPaymentAddress> ivkMap;
+    // Sapling ivk -> list of addrs map
+    std::map<libzcash::SaplingIncomingViewingKey, std::vector<libzcash::SaplingPaymentAddress>> ivkMap;
     for (const auto & addr : addresses) {
-        if (boost::get<libzcash::SaplingPaymentAddress>(&addr) != nullptr) {
-            auto saplingAddr = boost::get<libzcash::SaplingPaymentAddress>(addr);
+        auto saplingAddr = boost::get<libzcash::SaplingPaymentAddress>(&addr);
```

Comment:
gratuitous change (calling `get()` only once instead of twice), not for performance but simpler code (although this single trivial change did reduce the size of the object file by 8k bytes!)